WeakAurasOptions/AuthorOptions.lua

```diff
@@ -1100,9 +1174,109 @@ typeControlAdders = {
         get = get(option, "size"),
-        set = set(data, option, "size"),
+        set = function(_, value)
+          for id, optionData in pairs(option.references) do
+            local childOption = optionData.options[optionData.index]
+            local childData = optionData.data
+            if childOption.nameSource == -1 then
+              if value < childData.size then
+                for i = value + 1, childData.size do
+                  childOption.entryNames[i] = nil
+                end
+              else
+                for i = childData.size + 1, value do
+                  childOption.entryNames[i] = L["Entry %i"]:format(i)
+                end
+              end
+            end
+            childOption.size = value
+            WeakAuras.Add(childData)
+          end
+          WeakAuras.ReloadTriggerOptions(data)
+        end,
         disabled = function() return option.limitType == "none" end,
       }
+      args[prefix .. "hideReorder"] = {
+        type = "toggle",
+        name = name(option, "hideReorder", L["Disallow Entry Reordering"]),
+        desc = desc(option, "hideReorder"),
+        order = order(),
+        width = WeakAuras.normalWidth,
+        get = function()
+          return option.hideReorder or option.nameSource == -1
+        end,
+        set = set(data, option, "hideReorder"),
+        disabled = function()
+          return option.nameSource == -1
+        end,
+      }
+      local nameSources = CopyTable(WeakAuras.array_entry_name_types)
+      local validNameSourceTypes = WeakAuras.name_source_option_types
+      if option.limitType ~= "fixed" then
+        nameSources[-1] = nil
+      end
+      for subIndex, subOption in ipairs(option.subOptions) do
+        if validNameSourceTypes[subOption.type] then
+          local allShareThisOption = true
+          for id in pairs(option.references) do
+            if not subOption.references[id] then
+              allShareThisOption = false
+              break
+            end
+          end
+          if allShareThisOption then
+            nameSources[subIndex] = subOption.key
+          end
+        end
+      end
+      args[prefix .. "nameSource"] = {
+        type = "select",
+        name = name(option, "nameSource", L["Entry Name Source"]),
+        desc = desc(option, "nameSource"),
+        order = order(),
+        values = nameSources,
+        width = WeakAuras.doubleWidth,
+        get = function()
+          return option.nameSource or 0
+        end,
+        set = function(_, value)
+          for id, optionData in pairs(option.references) do
+            local childOption = optionData.options[optionData.index]
+            local childData = optionData.data
+            if (value == -1) ~= (childOption.nameSource == -1) then
+              if value == -1 then
+                local entryNames = {}
+                for i = 1, #childOption.subOptions do
+                  entryNames[i] = L["Entry #i"]:format(i)
```

Comment:
This code didn't work for me.

That is, with

!nsvZYTTnq4NMESDKuStDp6FKt0mkuUc0DC7bidsUKe1qaSaGYw(GF27UlOKODoKmdhcYf7)733kNkZLITsrf9GF2yCpllKIxwvxhGOmBIuSF03kBzRZFNtBJsX1ZZYNVwkkDotL7zR4zDhiV6KG5vna5nDPZsYJEDtd4dyKY4ipiG(Sck6RRZ3JEq815lV727xI3Z)Q69QzsrOdmMfvK1yWd9fWoWgfOv6xKInxFPiFJi)Y1OZ6TAuLoJAp6CHvTfgSInbdaOmX2to5opWorC38Llj96TNsTijqvg17G8KWV5Qaz2VoDcDd2kaVvz(lSW0yDMn7mP4ja6Uet4Y4Avu7OMGYQ3sFBPQnev(ix2yTrc3Kk1aGTkSehQCRZcum2Q02FETR1wDO9Nw)i18dQiPlpUiqW1oJZNMt0Zm(9N43NrVttG1qd69rZt4L4MqRch(pmc1me9(c6(bTsF(lHHFl)XXJ1R2rZVB96xF7X)SxvH5S6ThZFFShN8tyNnHD2Kp4SK2)94mLfdM6bi(L3NVAq6oDqxyy89XmjbxxDF(YfzZhK)okYISS5R3C1Q88vFB9IV818duOVlO)BFiQR3FIv5DrCEKZTPSvK7tw(WhTKseH(veroDwAC2c6MwuJpJirJtXmBcmYK9EtuFavxAuHW3jnqo7dcjkqFedpJ9PEGRC3be)ftMn58eUj1rk6JrNDfQasbhUOvx(KfWWHdawY4cXZaPKX82cCxsfYq5sOpaNGKfjUyc1DKhng1ZCsD8yn(Jrw9AA)35MLFPR4HfU)rkE152Yjgf4XZkLPRvXLWHLC0e6gDqHyJko76JoQbrwkkvgG1MIqg8mXbpcUomPR94ckCxicLzDtZ5BjPPoI461ZNNLC5TE4)6bB5EmD(TzNt5HTw3qndxh1xMEGngPILp(u64S0X5PJpNo(90XfPJ)iDGB2oo21wAoddLgMyR6IVJ0J86tIIhy7nExFhp7UbcLK10AyHR3xcCNCyapL1bNTgvxGrwpb7pvlSvifJ)9ThrbT6kyn48v4YzuBJEREGhIRWXrqkYjjkVNWFLh8EXa4oJxDhtD0LGTHZKjCZSspQw471iSL6q8FxLICbtuKIBGDsXo04ZMrqj5))

switching to Fixed Names didn't fill in the entryNames as this code appears to do

Reply:
Apparently that's because there's no sub option, but even with a subOption, that looks incorrect in two ways:

It should use childOption.size, it should use %i in the string